mlflow/entities/trace_info_v3.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass, field
+from typing import Any, Optional
+
+from google.protobuf.duration_pb2 import Duration
+from google.protobuf.json_format import MessageToDict
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from mlflow.entities._mlflow_object import _MlflowObject
+from mlflow.entities.assessment import Assessment
+from mlflow.entities.trace_location import TraceLocation
+from mlflow.entities.trace_state import TraceState
+from mlflow.entities.trace_status import TraceStatus
+from mlflow.protos import databricks_trace_server_pb2 as pb
+
+
+@dataclass
+class TraceInfoV3(_MlflowObject):
+    trace_id: str
+    trace_location: TraceLocation
+    request_time: int
+    state: TraceState
+    request: Optional[str] = None
+    response: Optional[str] = None
+    client_request_id: Optional[str] = None
+    execution_duration: Optional[int] = None
+    trace_metadata: dict[str, str] = field(default_factory=dict)
+    tags: dict[str, str] = field(default_factory=dict)
+    assessments: list[Assessment] = field(default_factory=list)
+
+    def to_dict(self) -> dict[str, Any]:
+        return MessageToDict(self.to_proto(), preserving_proto_field_name=True)
+
+    @classmethod
+    def from_dict(cls, d: dict[str, Any]) -> "TraceInfoV3":
+        d = d.copy()
+        if assessments := d.get("assessments"):
+            d["assessments"] = [Assessment.from_dictionary(a) for a in assessments]
+
+        if trace_location := d.get("trace_location"):
+            d["trace_location"] = TraceLocation.from_dict(trace_location)
+
+        if state := d.get("state"):
+            d["state"] = TraceState(state)
+
+        return cls(**d)
+
+    def to_proto(self):
+        from mlflow.tracing.constant import MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS
+
+        request_time = Timestamp()
+        request_time.FromMilliseconds(self.request_time)
+        execution_duration = None
+        if self.execution_duration is not None:
+            execution_duration = Duration()
+            execution_duration.FromMilliseconds(self.execution_duration)
+        return pb.TraceInfo(
+            trace_id=self.trace_id,
+            client_request_id=self.client_request_id,
+            trace_location=self.trace_location.to_proto(),
+            request=self.request,
+            response=self.response,
```

Comment:
Should we rename these to `request_preview` and `response_preview` as we are doing on the Mlflow backend to reflect that they can be truncated?

Reply:
Sounds good. I'll make this change in this PR.

Reply:
@ShaylanDias Filed https://github.com/mlflow/mlflow/pull/15328 to export `request_preview` and `response_preview` to OSS.